pkg/strategy/strategy.go

```diff
@@ -209,3 +211,69 @@ func (s S) Secrets() []corev1.Secret {
 func (s S) Dependencies() []batchv1.Job {
 	return s.dependencies
 }
+
+// All returns the list of all objects for this strategy
+func (s S) All() []runtime.Object {
```

Comment:
It should not fail, unless the object also has a namespace, but it might cause additional retries. I found a nice resource with an order of objects here: https://github.com/kubernetes-sigs/kustomize/blob/master/api/resid/gvk.go#L77-L103 so we should use that.